common/src/main/java/io/seata/common/thread/NamedThreadFactoryEx.java

```diff
@@ -0,0 +1,90 @@
+/*
+ *  Copyright 1999-2019 Seata.io Group.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+package io.seata.common.thread;
+
+import io.netty.util.concurrent.FastThreadLocalThread;
+
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ThreadFactory;
+import java.util.concurrent.atomic.AtomicInteger;
+
+/**
+ * Based on NamedThreadFactory
+ *
+ * @author jimin.jm @alibaba-inc.com
+ * @author ggndnn
+ */
+public class NamedThreadFactoryEx implements ThreadFactory {
```

Comment:
How about extend NamedThreadFactory directly?

Reply:
It's not easy to implement NamedThreadFactoryEx by extending NamedThreadFactory. I didn't mean to change NamedThreadFactory before, after your reminder, I'd like to modify NamedThreadFactory directly, just add a counter for every prefix.

Reply:
Committed.

Reply:
actually i mean you can use NamedThreadFactoryEx to extend NamedThreadFactory,and override the newThread method.

Reply:
The code you've committed previously was OK to me.what i mean is you can change the base class to NamedThreadFactory instead of implementing threadFactory directly.so,we can know this NamedThreadFactoryEx is only a extension for NamedThreadFactory for testing purpose.

Reply:
And applying this counter to all places will bring performance costs.

Reply:
And applying this counter to all places will bring performance costs.

Reply:
I understood your meaning. Your suggestion reminded me. The previous code was not so correct, the prefix counter should not increase in newThread but in constructor, so there is no performance cost. I think no harm done to add a prefix counter to NamedThreadFactory, just like java.util.concurrent.DefautThreadFactory's pool counter, not only for test, it's helpful in debugging thread related issue. What do you think?

Reply:
make sense.